salt/metalk8s/repo/configured.sls

```diff
@@ -48,3 +48,10 @@ Generate container registry configuration:
     - backup: false
     - defaults:
         archives: {{ archives }}
+
+{% set removable_archives = salt.metalk8s.get_removable_archives() %}
+{% for env in removable_archives %}
+Remove config for env {{ env }}:
+    file.absent:
+      - name: {{ salt.file.join(repo.config.directory, '99-' ~ env ~ '-registry.inc') }}
+{% endfor %}
```

Comment:
Again, to me we manage the full `repo.config.directory` so if we don't know a file in this directory we just remove it, since it's a bit complicated with solutions I suggest just checking all `99-metalk8s-*` we remove all files that are not expected

Reply:
:thinking: what if the Solution is called `metalk8s-solution-example`..? I'll go with a regex such as `99-metalk8s-[0-9].*`, should be enough.

Also, this state is a bit strange, we're not really managing all the includes, only adding one for the current version :confused:... I'll leave it like that but still confusing IMHO

Reply:
:thinking: what if the Solution is called `metalk8s-solution-example`..? I'll go with a regex such as `99-metalk8s-[0-9].*`, should be enough.

Also, this state is a bit strange, we're not really managing all the includes, only adding one for the current version :confused:... I'll leave it like that but still confusing IMHO

Reply:
Maybe solutions shouldn't have the "99-" prefix but another one, or yes just a simple regex

> Also, this state is a bit strange, we're not really managing all the includes, only adding one for the current version confused... I'll leave it like that but still confusing IMHO

That is my point, to me, we may want to change this state so that it basically "computes" the full directory content